### PR TITLE
Add `createProxy` function for storing observable objects.

### PR DESCRIPTION
**Pull request by dlongley**

### CODE REVIEW

File: createProxy.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        if(!proxy) {
+          proxy = new Proxy(value, handler);
+          map.set(value, proxy);
+        }
+        return proxy;
+      }
+      return value;
+    },
+    set(target, key, value) {
+      if(!target.hasOwnProperty(key)) {
```
Comment (mattcollier) [inline createProxy.js:25]: Should `tarket[key] = value;` occur in either case?  Here's a repl.it to illustrate my question.  https://repl.it/repls/ComposedChocolateApplicationprogrammer

Comment (dlongley) [reply createProxy.js:25]: It's up to the `setHandler` to set in that case:

https://repl.it/repls/ImmaculateGoldBooleanvalue

Comment (mattcollier) [reply createProxy.js:25]: OK, so it is the case that the expected behavior is that `setHandler` fires only when new properties are added to an object, but not when existing properties are updated (as demonstrated in the REPL)?

Comment (dlongley) [reply createProxy.js:25]: Yes... and perhaps we should do `{addProperty, setProperty}` options instead ... and allow for them to have defaults.

Comment (dlongley) [reply createProxy.js:25]: I pushed a change to support two handlers with defaults.

**Comment by mattcollier** [comment]
Just thought I would look quickly to see how others are doing this.  

Without making comment on the overall design of this module, it does seem to cover a lot of ground including, for instance, special handling of Arrays.  Is that level of functionality required here?

https://github.com/anywhichway/proxy-observe/blob/master/index.js

There are more than a few other implementation as well:

https://www.npmjs.com/search?q=proxy%20observable&ranking=popularity

**Comment by dlongley** [comment]
The `observe` API and pattern is obsolete and has been removed from browsers and node. We don't want anything that brings it back or polyfills it:

> This feature is obsolete. Although it may still work in some browsers, its use is discouraged since it could be removed at any time. Try to avoid using it.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/observe

**Comment by dlongley** [comment]
> ... it does seem to cover a lot of ground including, for instance, special handling of Arrays. Is that level of functionality required here?

We don't have a need for it at this time. I'm also not sure if it's even a good idea to support (some kind of special handling).

**Comment by dlongley** [comment]
Yeah, just checked ... we don't need any special handling of arrays, what we're doing already covers them.

**Review by mattcollier — APPROVED** [review]
Thanks for answering my questions.

**Pull request merged**